saleor/product/models.py

```diff
@@ -238,11 +238,13 @@ def display_variant(self, attributes=None):
                 ['%s: %s' % (smart_text(attributes.get(id=int(key))),
                              smart_text(value))
                  for (key, value) in values.items()])
-        else:
-            return smart_text(self.sku)
+        return ''
 
     def display_product(self):
-        return '%s (%s)' % (smart_text(self.product), smart_text(self))
+        variant_display = smart_text(self)
+        if variant_display:
+            return '%s (%s)' % (smart_text(self.product), smart_text(self))
+        return '%s' % (smart_text(self.product),)
```

Comment:
Instead of calling `smart_text` four times, we could assign the produced string to a variable and call `smart_text` in the last, single return statement.

Reply:
Do we actually need to use `smart_text`? I think it was mostly used to hide the differences between Python 2 and 3?